nuxeo/templates/deployment.yaml

```diff
@@ -72,6 +72,12 @@ spec:
             mountPath: /var/log/nuxeo
           - name: binaries
             mountPath: /binaries
+{{- if or .Values.elasticsearch.deploy .Values.tags.elasticsearch }}
+      initContainers:
+      - name: init-elasticsearch
+        image: busybox
+        command: ['sh', '-c', 'until wget -O/dev/null -q {{ .Release.Name }}-elasticsearch-client:9200 2>/dev/null; do echo waiting for {{ .Release.Name }}-elasticsearch-client; sleep 2; done;']
```

Comment:
rather look at a health check status for effective availability (also usually more efficient and reliable)
https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-health.html

Reply:
I think here we just want to check for the dns and not introduce dependency on the way the service is configured (port or protocol or health)
a simple nslookup is enough, the healthness is testd by the livenesprobe of the elasticsearch service that is using the es health status.
note also that check using busybox image and nslookup requires specific version to work properly (1.28)
https://github.com/docker-library/busybox/issues/48 -> https://github.com/kubernetes/website/pull/9901

Reply:
1) nslookup is not practical due to the service being up and resolvable even though the pods are not in a ready state.
2) as for the health, yes we could do that but once Nuxeo has access to ES it can read the cluster status and will mention it in its logs.

Reply:
yes so it sucks, k8s expose resolve the service name while it is not READY,
the readinessprobe logic must be duplicated in the nuxeo dep initContainers and using es health state is much better than letting nuxeo wait during start, this will fail faster in case of pb and we will not run into nuxeo es timeout

Reply:
So at the end it looks that K8s service ports are open only once the service is ready (the readiness probe of the service is successful), so we can just use "nc -w1 service port" to wait for the service, see next pr.

Reply:
https://github.com/nuxeo/nuxeo-helm-charts/pull/9